### PR TITLE
[jsk_tools] Add node to publish diagnostics based on topic and node status

### PR DESCRIPTION
**Pull request by 708yamaguchi**
This node checks topic and node status and publishes `/diagnostics` topic.

I refine https://github.com/jsk-ros-pkg/jsk_robot/pull/1460 by using `sanity_lib.py`.
https://github.com/jsk-ros-pkg/jsk_common/blob/a3d26ac5243c108cec25d4108d93b3f0fd57c503/jsk_tools/src/jsk_tools/sanity_lib.py

```
$ rostopic echo /diagnostics
header: 
  seq: 1
  stamp: 
    secs: 1652277254
    nsecs: 167789936
  frame_id: ''
status: 
  - 
    level: 0
    name: "/respeaker_node"
    message: "Node is alive"
    hardware_id: ''
    values: []
  - 
    level: 0
    name: "/kinect_head/kinect_head_nodelet_manager"
    message: "Node is alive"
    hardware_id: ''
    values: []
---
header: 
  seq: 2
  stamp: 
    secs: 1652277254
    nsecs: 601408004
  frame_id: ''
status: 
  - 
    level: 0
    name: "/kinect_head/rgb/image_raw"
    message: "Topic is published"
    hardware_id: ''
    values: []
  - 
    level: 0
    name: "/tf"
    message: "Topic is published"
    hardware_id: ''
    values: []
```

In the future, I think robot speaks `/diagnostics` topic with this script. 
https://github.com/jsk-ros-pkg/jsk_common/pull/1607

### CODE REVIEW

File: jsk_tools/src/sanity_diagnostics.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                state_msg.level = DiagnosticStatus.OK
+                state_msg.message = 'Node is alive'
+            else:
+                state_msg.level = DiagnosticStatus.ERROR
+                state_msg.message = 'Node is dead'
+            status_array.append(state_msg)
+        diag_array_msg.status = status_array
+        self.pub.publish(diag_array_msg)
+
+    def checkTopic(self):
```
Comment (iory) [inline jsk_tools/src/sanity_diagnostics.py:62]: I prefer `check_topic`

File: jsk_tools/src/sanity_diagnostics.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      - /kinect_head/rgb/image_raw
+      - /tf
+    nodes:
+      - /kinect_head/kinect_head_nodelet_manager
+      - /respeaker_node
+    """
+    def __init__(self):
+        yaml_path = rospy.get_param(
+            '~sanity_targets', "/var/lib/robot/sanity_targets.yaml")
+        if os.path.exists(yaml_path):
```
[the remark this answers is not in the document]
Comment (708yamaguchi) [reply jsk_tools/src/sanity_diagnostics.py:29]: Thank you!

I fixed in
[`7bc7d31` (#1727)](https://github.com/jsk-ros-pkg/jsk_common/pull/1727/commits/7bc7d31e4273a1a81d7349a9e55f095b5095059d)

File: jsk_tools/src/sanity_diagnostics.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                rospy.loginfo(sanity_targets)
+            self.pub = rospy.Publisher(
+                '/diagnostics', DiagnosticArray, queue_size=10)
+            self.timer = rospy.Timer(rospy.Duration(1), self.check_sanity)
+
+    def check_sanity(self, event):
+        self.checkNode()
+        self.checkTopic()
+
+    def checkNode(self):
```
[the remark this answers is not in the document]
Comment (708yamaguchi) [reply jsk_tools/src/sanity_diagnostics.py]: Thank you!

I fixed in
[`2dcf164` (#1727)](https://github.com/jsk-ros-pkg/jsk_common/pull/1727/commits/2dcf164b2e5c72196f3e720d13a96f12ed2afd59)

File: jsk_tools/src/sanity_diagnostics.py (hunk 1 above)
Comment (708yamaguchi) [reply jsk_tools/src/sanity_diagnostics.py]: Thank you!

I fixed in
[`2dcf164` (#1727)](https://github.com/jsk-ros-pkg/jsk_common/pull/1727/commits/2dcf164b2e5c72196f3e720d13a96f12ed2afd59)

File: jsk_tools/src/sanity_diagnostics.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    """
+    def __init__(self):
+        yaml_path = rospy.get_param(
+            '~sanity_targets', "/var/lib/robot/sanity_targets.yaml")
+        yaml_path = '/home/naoya/sanity_targets.yaml'
+        duration = rospy.get_param('~duration', 60)
+        if os.path.exists(yaml_path):
+            # Get sanity target topic and node names
+            with open(yaml_path) as yaml_f:
+                sanity_targets = yaml.load(yaml_f)
```
Comment (iory) [inline jsk_tools/src/sanity_diagnostics.py:34]: `yaml.load(yaml_f, Loader=yaml.SafeLoader)` is better for `python3`.

**Review by iory — CHANGES_REQUESTED** [review]

**Comment by iory** [comment]
This looks good!
Could you add a test and sample like https://github.com/jsk-ros-pkg/jsk_common/pull/1711/files

File: jsk_tools/src/sanity_diagnostics.py (hunk 4 above)
Comment (708yamaguchi) [reply jsk_tools/src/sanity_diagnostics.py]: Thank you.

I use `yaml.SafeLoader`
[`dcce522` (#1727)](https://github.com/jsk-ros-pkg/jsk_common/pull/1727/commits/dcce522f624da5b1b6df2e154b9c5af53cc55bd8)

**Comment by 708yamaguchi** [comment]
Thank you for your review.

I add sample
```
roslaunch jsk_tools sample_sanity_diagnostics.launch
```
and test
```
rostest jsk_tools test_sanity_diagnostics.test 
```

Result:
sanity_diagnostics

**Comment by 708yamaguchi** [comment]
Sorry for force-push.
I summerize commits 23 -> 6.

**Comment by iory** [comment]
This looks good.
https://github.com/jsk-ros-pkg/jsk_common/runs/6422833587?check_suite_focus=true#step:7:9433
Please fix the test error.

**Comment by 708yamaguchi** [comment]
Thank you!

I updated the code and fixed the test error.

**Pull request merged**